scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -106,42 +108,46 @@ class AstNamerMacros(val c: Context) extends Reflection with CommonNamerMacros {
         val fieldChecks = checkFieldsBuilder.result()
         val parentChecks = checkParentsBuilder.result()
 
-        stats1 ++= otherDefns
-
-        val binaryCompatAbstractFields = binaryCompatVars.flatMap(declareGetterSetter)
+        val binaryCompatAbstractFields = binaryCompatVars.flatMap { vr =>
+          val name = vr.name
+          val tpt = vr.tpt
+          stats1 += vr
+          stats1 += q"override def ${getterName(name)}: $tpt = this.$name"
+          stats1 += q"""
+            override def ${setterName(name)}($name : $tpt): Unit = {
+              val node = this
+              ${storeField(name)}
+            }
+          """
+          List(
+            declareGetter(name, tpt, Modifiers(DEFERRED, typeNames.EMPTY, vr.mods.annotations)),
+            q"def ${setterName(name)}($name : $tpt): Unit"
+          )
+        }
         istats1 ++= binaryCompatAbstractFields
-        stats1 ++= binaryCompatVars.flatMap(defineGetterSetter)
 
+        stats1 ++= otherDefns
         stats1 ++= imports
 
         // step 3: calculate the parameters of the class
         val paramss = rawparamss
 
         // step 4: turn all parameters into vars, create getters and setters
-        def internalize(name: TermName) = TermName("_" + name.toString)
         val fieldParamss = paramss
-        val fieldParams = fieldParamss.flatten.map(p => (p, p.name.decodedName.toString))
-        istats1 ++= fieldParams.map({ case (p, _) =>
-          var getterAnns = List(q"new $AstMetadataModule.astField")
-          if (p.mods.annotations.exists(_.toString.contains("auxiliary")))
-            getterAnns :+= q"new $AstMetadataModule.auxiliary"
```

Comment:
So `auxiliary` was never used? Maybe remove it altogether?

Reply:
i assumed that it was looking for AstMetadata.auxilliary and not just any old annotation, and simply kept all annotations instead.